proto/src/determined/checkpoint/v1/checkpoint.proto

```diff
@@ -50,9 +49,9 @@ message CheckpointTrainingMetadata {
     json_schema: { required: [] }
   };
   // The ID of the trial that created this checkpoint.
-  google.protobuf.Int32Value trial_id = 1;
+  optional int32 trial_id = 1;
   // The ID of the experiment that created this checkpoint.
-  google.protobuf.Int32Value experiment_id = 2;
```

Comment:
question: I'm not sure how the proto files work -- are these changes auto-generated?

Reply:
No, these changes are not auto-generated. But this proto file will create a pb.go file which is autogenerated. I have made this change here because the protobuf.Int32Value type was very inconvenient to scan into. I asked the backend-eng-group if it was alright to make this change as well (thread: https://hpe-aiatscale.slack.com/archives/C02PV33GSN5/p1707247966754879)